### DISCUSSION TITLE
Automatic noise level sensing

### DISCUSSION THREAD

**Discussion by nzreuben** [Breakthrough/DVR-Scan#166] [Ideas]
I am proving that the traffic noise outside my house is too loud.  DVR-Scan is awesome for breaking down all the vehicle activity into bite-sized videos.  But then I have to go through them all manually to find the 5-10% that are super loud.  So it would be great if you could add a noise filter that only saves all videos where the sound is greater than 80DB for example - and also goes on for more than say 3 seconds.  I think it would be a great addition to an already excellent tool.   It could no doubt be applied to many other applications too.

**Comment by Breakthrough** [comment]
It looks like you can [use ffmpeg's `volumedetect` filter](https://trac.ffmpeg.org/wiki/AudioVolume#PeakandRMSNormalization) for this purpose.  You could write a script that post-processes the DVR-Scan output to find only clips that have a peak volume above a given threshold, e.g.:

```python
import glob
import os
import shutil
import sys
import subprocess

IDENTIFIER = "max_volume: "

if len(sys.argv) != 4:
    print(f"Usage: {sys.argv[0]} path/to/videos path/to/output 100.5")
    exit(1)
source_dir = os.path.abspath(sys.argv[1])
dest_dir = os.path.abspath(sys.argv[2])
threshold = float(sys.argv[3])
os.makedirs(dest_dir, exist_ok=True)
for file in glob.glob(os.path.join(source_dir, "*")):
    output = subprocess.run(["ffmpeg", "-v", "info", "-i", file, "-filter:a", "volumedetect", "-f", "null", "-"], capture_output=True, text=True)
    if output.returncode != 0:
        continue
    start = output.stderr.find(IDENTIFIER)
    if start < 0:
        continue
    start += len(IDENTIFIER)
    end = output.stderr.find(" ", start)
    volume = float(output.stderr[start:end])
    print(f"{os.path.basename(file)} volume: {volume} dB")
    if volume > threshold:
        shutil.copyfile(file, os.path.join(dest_dir, os.path.basename(file)))
```

If you're unable to get something working feel free to reach out or file a feature request for this.

**Reply by nzreuben** [reply]
Thanks so much @Breakthrough  It's such cool sand powerful software.  If you could supply what I need to add to the command line that would be amazing.   Thanks!  🤞👍

**Reply by Breakthrough** [reply]
Updated the comment with a simple Python script to do this.  You can run that the same as DVR-Scan and it will copy those motion events with audio levels greater than the threshold you provide.  It could be done as well in bash with a regex, but wanted to leave this here if someone wants to take this on and add it to DVR-Scan.

If we add this it should probably be a config file only option to avoid adding more complexity to the command line interface.  If there are more folks who want this as an official feature I'm happy to prioritize it, but the above Python script should do the same job.

**Reply by nzreuben** [reply]
So I simply add it to the config file I am using?  I can do that!

**Reply by Breakthrough** [reply]
> So I simply add it to the config file I am using? I can do that!

This is a separate Python script, you'll need to save it to a file like "filter_volume.py" and run it with `python3 filter_volume.py` once you're done with DVR-Scan.
